### ISSUE TITLE
Extra test for you.

### ISSUE THREAD

**Issue by rdebath** [mgthomas99/picobrain#1]
This should print "Hello World!".
From reading your code I'm guessing it won't work.
``` bf
+[>[<->+[>+++>[+++++++++++>][>]-[<]>-]]++++++++++<]>>
>>>>----.<<+++.<-..+++.<-.>>>.<<.+++.------.>-.<<+.<.
```

**Comment by mgthomas99** [comment]
Hi @rdebath, thanks for taking an interest in the project!

I have managed to find the cause of the bug, and the latest version of the package, `2.0.1`, fixes it. The bug was being caused due to the way I was iterating over nested for-loops, which your script contains.

The script you posted now works. I've also added another test which specifically tests nested loops. Thank you for finding that bug!

**Comment by rdebath** [comment]
`i+=n>0?1:n<0?-1:1` --> `i+=n>=0?1:-1`

**Comment by mgthomas99** [comment]
@rdebath I don't know how I missed that, but thanks! I've fixed it in the latest version ([`2.0.2`](https://github.com/mgthomas99/picobrain/blob/master/doc/CHANGELOG.md#202)).

Would you mind if I add you as a contributor to the package?

**Comment by rdebath** [comment]
I'm fine with that.
